### PR TITLE
Reduce code duplication in testing defaults handling

### LINKED ISSUE #3414: [REFACTOR] Solve duplication of `TestBot.test_defaults_handling` and `check_defaults_handling`
Currently `TestBot.test_defaults_handling` calls `check_defaults_handling` for every ext bot method. Both these functions do a similar thing and have common code. We can check if this can be simplified or merged.

### PR DESCRIPTION
**Pull request by Bibo-Joshi**
Closes #3414. I double checked that tests do fail if any of the defaults-handling logic in `tg.Bot` is removed.

* Finds a solution to not having to call `get_me` after each assertion or even after checking `get_me` via mocking it.
* Introduces a new directory `tests/auxil` with a file `bot_methods_checks.py` in it where I moved the respective functions from `conftest.py` to. If this finds approval, I would move all other non-fixture things from `conf.py` into separate files within `tests/auxil`.

### CODE REVIEW

**Review by harshil21 — COMMENTED** [review]
I'm definitely for splitting up non-fixture things from conftest to this new directory. In fact, can you also extract the nested functions inside `check_defaults_handling` into new functions (maybe in the same file)? Would make the code in there more readable at least

File: tests/test_bot.py [hunk 1]
```diff
@@ -452,74 +444,19 @@ async def test_defaults_handling(
         if bot_method_name.lower().replace("_", "") == "getupdates":
```
[the remark this answers is not in the document]
Comment (Bibo-Joshi) [reply tests/test_bot.py:444]: I vaguely recall that having something to do with multiple arguments called `timeout` back in the day … let's see if I can test it as well

File: tests/auxil/bot_method_checks.py [hunk 2]
```diff
@@ -38,6 +38,11 @@
 from telegram.constants import InputMediaType
 from telegram.ext import Defaults, ExtBot
 from telegram.request import RequestData
+from tests.auxil.object_conversions import env_var_2_bool
+
+TEST_WITH_OPT_DEPS = env_var_2_bool(os.getenv("TEST_WITH_OPT_DEPS", True))
```
[the remark this answers is not in the document]
Comment (Bibo-Joshi) [reply tests/auxil/bot_method_checks.py]: reverting since this leeds to nasty import errors

**Comment by Bibo-Joshi** [comment]
> In fact, can you also extract the nested functions inside `check_defaults_handling` into new functions (maybe in the same file)? Would make the code in there more readable at least

Those functions are nested because they work with several of the variables defined within `check_defaults_handling`/`check_shortcut_call`. Extracting those methods would mean that I'd have to add several more arguments to them & set them with `functools.partial`. I don't see a compelling benefit of that tbh …

**Review by harshil21 — APPROVED** [review]
Great, I can verify if this works only after this is merged to master, but the tests passing here are a good indication it will.

**Pull request merged**